### PR TITLE
Concat flag

### PR DESCRIPTION
**Pull request by AdrianTeng**
@achamayou undo-ed #363 in #373, due to breakage in AHL internal test. Turns out the broken test is just copies of https://github.com/manahl/arctic/blob/master/tests/integration/store/test_version_store.py#L261, so I just need to apply the same test change.

### CODE REVIEW

File: arctic/store/_pandas_ndarray_store.py [hunk 1]
```diff
@@ -116,6 +116,24 @@ def get_info(self, version):
...
+    def read_segment_last_dt(self, version):
+        if 'segment_index' in version:
+            index = np.fromstring(decompress(version['segment_index']), dtype=INDEX_DTYPE)
+            dt_index = self._datetime64_index(index)
+            if dt_index:
+                return index[dt_index][-1]
+        return None
+
+    def slice_overlap_item_or_raise(self, item, previous_version, concat):
+        """If new item has overlap dt with previous version, keep only new bits if concat=True; raise if concat=False"""
```
Comment (achamayou) [inline arctic/store/_pandas_ndarray_store.py]: Maybe mention the type that's raised?

File: arctic/store/_pandas_ndarray_store.py [hunk 2]
```diff
@@ -116,6 +116,24 @@ def get_info(self, version):
         ret['dtype'] = ast.literal_eval(version['dtype'])
         return ret
 
+    def read_segment_last_dt(self, version):
```
Comment (achamayou) [inline arctic/store/_pandas_ndarray_store.py]: Since this is public, best to give it a docstring?

**Comment by achamayou** [comment]
Strange, it's the same transient error I've seen before as well:

`>       raise AssertionError(msg)
E       AssertionError: DataFrame.columns are different
E       
E       DataFrame.columns values are different (66.66667 %)
E       [left]:  Index(['a', 'c', 'b'], dtype='object')
E       [right]: Index(['a', 'b', 'c'], dtype='object')`

This only seems to happen on 3.5, and not every time.

**Comment by bmoscon** [comment]
@achamayou yes its transient, only on 3.5 and only on travis.

**Pull request merged**